apis/vmware/v1beta1/vspherecluster_types.go

```diff
@@ -30,6 +30,7 @@ const (
 
 // VSphereClusterSpec defines the desired state of VSphereCluster.
 type VSphereClusterSpec struct {
+	// +optional
```

Comment:
This aligns VSphereClusterSpec in supervisor API objects to the same object in the govimomi API.
(it makes it possible to not specify a ControlPlaneEndpoint in a VSphereClusterTemplate)